File: src/Postgres.ts
import {namespacePostgres} from "./namespace";
import * as k8s from "@pulumi/kubernetes"
import {mariaDbBackupSecret} from "./Secrets";

const dbPassword = process.env.CI_DB_PASSWORD
const dbUsername = process.env.CI_DB_USERNAME
const dbRootPassword = process.env.CI_DB_ROOT_PASSWORD;
 function createPostgres() {

  return new k8s.helm.v3.Chart("postgres", {
        chart: "postgresql",
        namespace: namespacePostgres.metadata.name,
        fetchOpts: {
          repo: "https://charts.bitnami.com/bitnami",
        },
        values: {
         "global": {
           "storageClass": "juice",
           "postgresql":{
             "auth": {
               "database": "directus",
               "username": dbUsername,
               "password": dbPassword,
               "postgresPassword": dbRootPassword
             }
           }
         }
        }
      }
  );
}

export const postgres = createPostgres()

File: modules/database/postgres/postgres.go
package postgres

import (
	helmv4 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/helm/v4"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func createPostgres() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := helmv4.NewChart(ctx, "nginx", &helmv4.ChartArgs{
			Chart: pulumi.String("nginx"),
			RepositoryOpts: &helmv4.RepositoryOptsArgs{
				Repo: pulumi.String("https://charts.bitnami.com/bitnami"),
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}
